docs/source/examples/mg-mode-part-i.ipynb

```diff
@@ -0,0 +1,707 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This Notebook illustrates the usage of OpenMC's multi-group calculational mode with the Python API. This example notebook creates and executes the 2-D [C5G7](https://www.oecd-nea.org/science/docs/2003/nsc-doc2003-16.pdf) benchmark model using the `openmc.MGXSLibrary` class to create the supporting data library on the fly."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generate MGXS Library"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.colors as colors\n",
+    "import numpy as np\n",
+    "\n",
+    "import openmc\n",
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We will now create the multi-group library using data directly from Appendix A of the C5G7 benchmark documentation. All of the data below will be created at 294K, consistent with the benchmark.\n",
+    "\n",
+    "This notebook will first begin by setting the group structure and building the data for UO2.\n",
+    "\n",
+    "*Note*: Most users familiar with deterministic transport libraries are already familiar with the idea of entering one scattering matrix for every order (i.e. scattering order as the outer dimension). However, the shape of OpenMC's scattering matrix entry is instead [Incoming groups, Outgoing Groups, Scattering Order] to best enable other scattering representations. We will follow the more familiar approach in this notebook, and then use numpy's `numpy.rollaxis` function to change the ordering to what we need (scattering order on the inner dimension).\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Create a 7-group structure with arbitrary boundaries (the specific boundaries are unimportant)\n",
+    "groups = openmc.mgxs.EnergyGroups(np.logspace(-5, 7, 8))\n",
+    "\n",
+    "uo2_xsdata = openmc.XSdata('uo2', groups)\n",
+    "uo2_xsdata.order = 0\n",
+    "\n",
+    "# When setting the data let the object know you are setting the data for a temperature of 294K.\n",
+    "uo2_xsdata.set_total([1.77949E-1, 3.29805E-1, 4.80388E-1, 5.54367E-1,\n",
+    "                      3.11801E-1, 3.95168E-1, 5.64406E-1], temperature=294.)\n",
```

Comment:
This appears in the C5G7 spec as transport cross section -- might be good to mention that. Also, probably good to point out that values are given in order of increasing group / decreasing energy.